app/controllers/registrations_controller.rb

```diff
@@ -558,12 +558,10 @@ def stripe_webhook
   end
 
   def payment_completion
-    registration = Registration.includes(:competition).find(params[:id])
-    @competition = registration.competition
-
     # Provided by Stripe upon redirect when the "PaymentElement" workflow is completed
     intent_id = params[:payment_intent]
     intent_secret = params[:payment_intent_client_secret]
+    @competition = params[:competition_id]
```

Comment:
...ehm??

Reply:
This is necessary due to a hen and egg problem, we can only get the registration out of the payment id, but we can't redirect on `unless stored_record.payment_intent?` because we don't know the competition_id yet. Instead I changed the url to be `competitions/:competition_id/payment-completion` (which I think makes more sense anyway) and then we can get the competition_id directly

Reply:
I renamed it to be clearer